content/en/hosting/4.x/_partial_migration_3x_docker_to_4x_k3s.md

```diff
@@ -0,0 +1,62 @@
+---
+toc_hide: true
+hide_summary: true
+---
+
+The hosting architecture differs entirely between CHT-Core 3.x and CHT-Core 4.x. When migrating from Docker Compose to K3s, specific steps are required using the [couchdb-migration](https://github.com/medic/couchdb-migration) tool. This tool interfaces with CouchDB to update shard maps and database metadata.
+
+{{% alert title="Note" %}}
+If after upgrading you get an error, `Cannot convert undefined or null to object` - please see [issue #8040](https://github.com/medic/cht-core/issues/8040) for a work around. This only affects CHT 4.0.0, 4.0.1, 4.1.0 and 4.1.1. It was fixed in CHT 4.2.0.
+{{% /alert %}}
+
+
+## Install Migration Tool
+```shell
+mkdir -p ~/couchdb-migration/
+cd ~/couchdb-migration/
+curl -s -o ./docker-compose.yml https://raw.githubusercontent.com/medic/couchdb-migration/main/docker-compose.yml
+docker compose up
+```
+
+## Set Up Environment Variables
+```shell
+# Replace with your actual CouchDB URL from the Docker Compose setup
+export COUCH_URL=http://<your-admin-user>:<your-admin-password>@<couchdb-host>:5984
+```
```

Comment:
We should standardize on using the same `<variable-name>`  for username and password.  As well, let's pull the code comment out to be text that's easier to read:

``````suggestion
## Set Up Environment Variables

Be sure to replace both `<admin-user>` and `<password>` with your actual username and password.  As well, update `<couchdb-host>` to the CouchDB URL from the Docker Compose setup:

```shell
export COUCH_URL=http://<admin-user>:<password>@<couchdb-host>:5984
```
``````